HeliumVotePlugin/components/LockTokensAccount.tsx

```diff
@@ -0,0 +1,410 @@
+import { daysToSecs } from '@utils/dateTools'
+import useVotePluginsClientStore from 'stores/useVotePluginsClientStore'
+import { LockCommunityTokensBtn } from './LockCommunityTokensBtn'
+import { LockTokensModal, LockTokensModalFormValues } from './LockTokensModal'
+import { useCreatePosition } from '../hooks/useCreatePosition'
+import { calcLockupMultiplier } from '../utils/calcLockupMultiplier'
+import { VotingPowerBox } from './VotingPowerBox'
+import { PositionCard } from './PositionCard'
+import useHeliumVsrStore from '../hooks/useHeliumVsrStore'
+import { PublicKey } from '@solana/web3.js'
+import { notify } from '@utils/notifications'
+
+export const LockTokensAccount: React.FC<{
+  tokenOwnerRecordPk: string | string[] | undefined
+  children: React.ReactNode
+}> = ({ tokenOwnerRecordPk, children }) => {
+  const {
+    mint,
+    realm,
+    realmTokenAccount,
+    realmInfo,
+    tokenRecords,
+    councilMint,
+    config,
+  } = useRealm()
+
+  const tokenOwnerRecordWalletPk = Object.keys(tokenRecords)?.find(
+    (key) => tokenRecords[key]?.pubkey?.toBase58() === tokenOwnerRecordPk
+  )
+
+  const [
+    currentClient,
+    vsrClient,
+    vsrRegistrar,
+  ] = useVotePluginsClientStore((s) => [
+    s.state.currentRealmVotingClient,
+    s.state.heliumVsrClient,
+    s.state.heliumVsrRegistrar,
+  ])
+  const { error, createPosition } = useCreatePosition()
+  const [isOwnerOfPositions, setIsOwnerOfPositions] = useState(true)
+  const [isLockModalOpen, setIsLockModalOpen] = useState(false)
+  const [
+    connected,
+    connection,
+    wallet,
+    { fetchRealm, fetchWalletTokenAccounts },
+  ] = useWalletStore((s) => [
+    s.connected,
+    s.connection.current,
+    s.current,
+    s.actions,
+  ])
+
+  const [
+    isLoading,
+    positions,
+    votingPower,
+    amountLocked,
+    getPositions,
+  ] = useHeliumVsrStore((s) => [
+    s.state.isLoading,
+    s.state.positions,
+    s.state.votingPower,
+    s.state.amountLocked,
+    s.getPositions,
+  ])
+
+  const handleGetPositions = useCallback(async () => {
+    try {
+      if (
+        config?.account.communityTokenConfig.voterWeightAddin &&
+        realm?.pubkey &&
+        wallet?.publicKey &&
+        vsrClient
+      ) {
+        await getPositions({
+          votingClient: currentClient,
+          realmPk: realm.pubkey,
+          communityMintPk: realm.account.communityMint,
+          walletPk: tokenOwnerRecordWalletPk
+            ? new PublicKey(tokenOwnerRecordWalletPk)
+            : wallet.publicKey,
+          client: vsrClient,
+          connection: connection,
+        })
+      }
+    } catch (e) {
+      notify({
+        type: 'error',
+        message: 'Unable to fetch positions',
+      })
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps -- TODO please fix, it can cause difficult bugs. You might wanna check out https://bobbyhadz.com/blog/react-hooks-exhaustive-deps for info. -@asktree
+  }, [])
```

Comment:
this looks dubious to me --
Firstly I notice this callback has no dependencies which seems like it could be a source of bugs. 
Secondly, if you use a callback like this with an effect, I think it's best to inline the callback inside of the effect. However, rather than using an effect, this seems like a perfect time to use the `react-async-hook` library.

Reply:
Good call out!